src/hotspot/cpu/x86/x86.ad

```diff
@@ -3676,6 +3676,7 @@ instruct vconvF2HF(vec dst, vec src) %{
 %}
 
 instruct vconvF2HF_mem_reg(memory mem, vec src) %{
+  predicate(Matcher::vector_length_in_bytes(n->in(3)->in(1)) >= 16);
```

Comment:
You can add an eligant prediction check like following instead of accesing bare inputs.

n->as_StoreVector()->memory_size() >= 16.

Reply:
We have used bare inputs at many places in the ad file in the predicate.

Reply:
I think its ok to use safe cast if its available and improve new code.

Reply:
@jatin-bhateja I have made this change. Please take a look.